app/src/main/java/com/github/se/polyfit/ui/components/selector/DateSelector.kt

```diff
@@ -0,0 +1,110 @@
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.rememberDatePickerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.github.se.polyfit.ui.theme.PrimaryPink
+import com.github.se.polyfit.ui.theme.PurpleGrey40
+import java.time.LocalDate
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun DateSelector(onConfirm: (LocalDate) -> Unit, modifier: Modifier = Modifier) {
+  var showDatePicker by remember { mutableStateOf(false) }
+
+  // I'm *pretty sure* this should select the right time and zone, but if test flakes this is why
+  val initialMillis = LocalDate.now().atStartOfDay().toInstant(ZoneOffset.UTC).toEpochMilli()
+  val datePickerState = rememberDatePickerState(initialSelectedDateMillis = initialMillis)
+
+  val selectedDate =
+      LocalDate.ofEpochDay(datePickerState.selectedDateMillis!! / (1000 * 60 * 60 * 24))
+  val dateString = selectedDate.format(DateTimeFormatter.ISO_LOCAL_DATE)
+
+  fun onConfirm() {
+    val date = LocalDate.parse(dateString, DateTimeFormatter.ISO_LOCAL_DATE)
+    onConfirm(date)
+    showDatePicker = false
+  }
+
+  Column(modifier = modifier.testTag("DateSelector")) {
+    Text(
+        text = "Add to Date",
+        style = MaterialTheme.typography.titleLarge,
+        color = PurpleGrey40,
+        modifier = Modifier.padding(16.dp, 0.dp).testTag("Title"))
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = Modifier.fillMaxWidth().testTag("DateBox")) {
+          TextButton(
+              onClick = { showDatePicker = true },
+              modifier = Modifier.testTag("ChangeDateButton")) {
+                FormatDate(dateString)
+              }
+
+          if (showDatePicker) {
+            DatePickerDialog(
+                onDismissRequest = { showDatePicker = false },
+                confirmButton = { TextButton(onClick = { onConfirm() }) { Text(text = "Done") } },
+                dismissButton = {
+                  TextButton(onClick = { showDatePicker = false }) { Text(text = "Cancel") }
+                },
+                modifier = Modifier.testTag("DatePickerDialog")) {
+                  DatePicker(state = datePickerState, modifier = Modifier.testTag("DatePicker"))
+                }
+          }
+        }
+  }
+}
+
+@Composable
+private fun FormatDate(dateString: String) {
```

Comment:
## Bug Report: App Crash on Date Input Manipulation

**Issue Summary:**  
The application crashes when a user attempts to delete part of the year component (or any parts of the date). This was observed when modifying the date "12/03/2024" to remove the last digit of the year.

**Steps to Reproduce:**
1. Open the date entry field.
2. Click on the pencil icon to input the data with the keyboard instead of the gui.
3. Delete the digit '4' from the year.

**Expected Result:**  
Disable the done button until the user entered a correctly formatted date.

**Actual Result:**  
The preview crashes.

**Video Proof:**  
[Screen_recording_20240421_111431.webm](https://github.com/swent-group10/polyfit/assets/100274930/0ce2f805-ad30-4adb-b441-f3d4506f69d0)


---

Reply:
Good find 🚀 Now users have to select a date from the dialog and can't input any invalid date